master/buildbot/db/pool.py

```diff
@@ -170,37 +170,30 @@ def __thd(self, with_engine, callable, args, kwargs):
                 arg = self.engine
             else:
                 arg = self.engine.contextual_connect()
-
             try:
                 try:
                     rv = callable(arg, *args, **kwargs)
                     assert not isinstance(rv, sa.engine.ResultProxy), \
                         "do not return ResultProxy objects!"
                 except sa.exc.OperationalError as e:
-                    text = e.orig.args[0]
-                    if not isinstance(text, basestring):
+                    if not self.engine.should_retry(e):
+                        log.err(e, 'Got fatal OperationalError on DB')
                         raise
-                    if "Lost connection" in text \
-                            or "database is locked" in text:
-
-                        # see if we've retried too much
-                        elapsed = time.time() - start
-                        if elapsed > self.MAX_OPERATIONALERROR_TIME:
-                            raise
-
-                        metrics.MetricCountEvent.log(
-                            "DBThreadPool.retry-on-OperationalError")
-                        log.msg("automatically retrying query after "
-                                "OperationalError (%ss sleep)" % backoff)
-
-                        # sleep (remember, we're in a thread..)
-                        time.sleep(backoff)
-                        backoff *= self.BACKOFF_MULT
-
-                        # and re-try
-                        continue
-                    else:
+                    elapsed = time.time() - start
+                    if elapsed > self.MAX_OPERATIONALERROR_TIME:
+                        log.err(e, 'Raising due to 1day delay on DB query retries')
```

Comment:
I'm assuming `self.MAX_OPERATIONALERROR_TIME` is one day here, but if that constant is changed then this message would be incorrect.  Maybe just reference the constant?